src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -274,6 +274,9 @@
          cacheVariableName <- options$cacheVariableNames[[resource]]
          cacheUrlName <- options$cacheUrlNames[[resource]]
          downloadResource <- options[[resource]]
+         resourceExtension <- sub("[^\\?]*([.][^.\\?]*)(\\?.*)?$", "\\1", basename(downloadResource), perl = TRUE)
```

Comment:
Thanks for pointing this out... however, it doesn't work for all the cases since it won't consider urls. Therefore, things like `tools::file_ext("test.txt?test=.xyz")` will return `"xyz"` or `tools::file_ext("test.txt?test")` returns `""`.